presidio-analyzer/tests/test_analyzer_engine.py

```diff
@@ -282,4 +289,12 @@ def test_when_allFields_is_true_and_entities_not_empty_exception(self):
         new_field.minScore = '0.5'
         with pytest.raises(ValueError):
             analyze_engine.Apply(request, None)
-    
\ No newline at end of file
+
+    def test_when_analyze_then_mltracer_has_value(self):
+        text = " Credit card: 4095-2609-9393-4932,  my phone is 425 8829090"
+        language = "en"
+        entities = ["CREDIT_CARD", "PHONE_NUMBER"]
+        analyzer_engine_with_spacy = AnalyzerEngine(self.loaded_registry, app_tracer=self.app_tracer)
+        results = analyzer_engine_with_spacy.analyze(self.unit_test_guid, text, entities, language, all_fields=False)
+        assert len(results) == 2
+        assert self.app_tracer.get_last_trace() is not None
```

Comment:
Better to test that it has all the different parts traced- NLP artifacts, AnalysisExplanations